engine/src/main/java/org/pentaho/di/job/entries/filesexist/JobEntryFilesExist.java

```diff
@@ -0,0 +1,264 @@
+import org.pentaho.di.job.JobMeta;
+import org.pentaho.di.job.entry.JobEntryBase;
+import org.pentaho.di.job.entry.JobEntryInterface;
+import org.pentaho.di.repository.ObjectId;
+import org.pentaho.di.repository.Repository;
+import org.pentaho.metastore.api.IMetaStore;
+import org.w3c.dom.Node;
+
+/**
+ * This defines a Files exist job entry.
+ *
+ * @author Samatar
+ * @since 10-12-2007
+ *
+ */
+
+public class JobEntryFilesExist extends JobEntryBase implements Cloneable, JobEntryInterface {
+  private static Class<?> PKG = JobEntryFilesExist.class; // for i18n purposes, needed by Translator2!!
+
+  private String filename; // TODO: looks like it is not used: consider deleting
+
+  private String[] arguments;
+
+  public JobEntryFilesExist( String n ) {
+    super( n, "" );
+    filename = null;
+  }
+
+  public JobEntryFilesExist() {
+    this( "" );
+  }
+
+  public void allocate( int nrFields ) {
+    arguments = new String[nrFields];
+  }
+
+  public Object clone() {
+    JobEntryFilesExist je = (JobEntryFilesExist) super.clone();
+    if ( arguments != null ) {
+      int nrFields = arguments.length;
+      je.allocate( nrFields );
+      System.arraycopy( arguments, 0, je.arguments, 0, nrFields );
+    }
+    return je;
+  }
+
+  public String getXML() {
+    StringBuilder retval = new StringBuilder( 30 );
+
+    retval.append( super.getXML() );
+    if ( parentJobMeta != null ) {
+      parentJobMeta.getNamedClusterEmbedManager().registerUrl( filename );
+    }
+    retval.append( "      " ).append( XMLHandler.addTagValue( "filename", filename ) );
+
+    retval.append( "      <fields>" ).append( Const.CR );
+    if ( arguments != null ) {
+      for ( int i = 0; i < arguments.length; i++ ) {
+        retval.append( "        <field>" ).append( Const.CR );
+        retval.append( "          " ).append( XMLHandler.addTagValue( "name", arguments[i] ) );
+        retval.append( "        </field>" ).append( Const.CR );
+        if ( parentJobMeta != null ) {
+          parentJobMeta.getNamedClusterEmbedManager().registerUrl( arguments[i] );
+        }
+      }
+    }
+    retval.append( "      </fields>" ).append( Const.CR );
+
+    return retval.toString();
+  }
+
+  public void loadXML( Node entrynode, List<DatabaseMeta> databases, List<SlaveServer> slaveServers,
+    Repository rep, IMetaStore metaStore ) throws KettleXMLException {
+    try {
+      super.loadXML( entrynode, databases, slaveServers );
+      filename = XMLHandler.getTagValue( entrynode, "filename" );
+
+      Node fields = XMLHandler.getSubNode( entrynode, "fields" );
+
+      // How many field arguments?
+      int nrFields = XMLHandler.countNodes( fields, "field" );
+      allocate( nrFields );
+
+      // Read them all...
+      for ( int i = 0; i < nrFields; i++ ) {
+        Node fnode = XMLHandler.getSubNodeByNr( fields, "field", i );
+
+        arguments[i] = XMLHandler.getTagValue( fnode, "name" );
+
+      }
+    } catch ( KettleXMLException xe ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)